src/PowerShellEditorServices/Analysis/AnalysisService.cs

```diff
@@ -283,11 +316,11 @@ public IEnumerable<string> GetPSScriptAnalyzerRules()
         /// </summary>
         public void Dispose()
         {
-            if (this._analysisRunspacePool != null)
+            if (_analysisRunspacePool != null)
             {
-                this._analysisRunspacePool.Close();
-                this._analysisRunspacePool.Dispose();
-                this._analysisRunspacePool = null;
+                _analysisRunspacePool.Close();
+                _analysisRunspacePool.Dispose();
+                _analysisRunspacePool = null;
```

Comment:
is the nulling needed here? I'm not sure.

Reply:
Well really we should implement the [Dispose pattern](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/dispose-pattern). I'm guessing the nulling is to allow GC to collect, but that's protecting against the scenario where the parent is disposed and the reference still carried around for a while. It doesn't hurt I guess.